lib/Platform/Intl/java/com/facebook/hermes/intl/PlatformDateTimeFormatterAndroid.java

```diff
@@ -204,8 +211,21 @@ else if (needTime)
   }
 
   @Override
-  public boolean isValidTimeZone(String timeZone) {
-    return TimeZone.getTimeZone(timeZone).getID().equals(timeZone);
+  public String normalizeValidTimeZone(final String timeZone) throws JSRangeErrorException {
+    Optional<String> normalizedValidTimeZone = allAvailableTimeZones.stream()
+    .filter(new Predicate<String>() {
+      @Override
+      public boolean test(String tz) {
+        return tz.compareToIgnoreCase(timeZone) == 0;
+      }
+    }).findAny();
+
+    if(!normalizedValidTimeZone.isPresent()) {
+      String errorMessage = timeZone + " is invalid timeZone";
+      throw new JSRangeErrorException(errorMessage);
+    }
+
+    return normalizedValidTimeZone.get();
```

Comment:
I think it would be easier to read and understand this code if it was just a for loop that returns as soon as it finds a match, but perhaps that's a matter of personal preference.

```suggestion
    for (String id : TimeZone.getAvailableIDs()){
      if(id.compareToIgnoreCase(timeZone) == 0){
        return id;
      }
    }
    throw new JSRangeErrorException(timeZone + " is an invalid timeZone");
```

Reply:
This is much cleaner, thank you 👏